grc/core/blocks/dummy.py

```diff
@@ -51,4 +55,6 @@ def add_missing_port(self, port_id, direction):
             self.sources.append(port)
         else:
             self.sinks.append(port)
+        self.rewrite()     # Make port visible
+        self.validate()    # Generate appropriate error messages     
```

Comment:
I am not sure if these should be placed here. Maybe better on the call-site in FlowGraph.import_data